packages/react/src/components/cookieConsent/cookieConsentController.ts

```diff
@@ -0,0 +1,239 @@
+import _pick from 'lodash.pick';
+import _isObject from 'lodash.isobject';
+import _isUndefined from 'lodash.isundefined';
+
+import { createCookieController } from './cookieController';
+
+export type ConsentList = string[];
+
+export type ConsentObject = { [x: string]: boolean };
+
+export type ConsentStorage = {
+  required: ConsentObject;
+  optional: ConsentObject;
+  unknown?: ConsentObject;
+};
+
+export type ConsentControllerProps = {
+  requiredConsents?: ConsentList;
+  optionalConsents?: ConsentList;
+  cookieDomain?: string;
+};
+
+export type ConsentController = {
+  getRequired: () => ConsentObject;
+  getOptional: () => ConsentObject;
+  update: (key: string, value: boolean) => void;
+  approveRequired: () => void;
+  approveAll: () => void;
+  rejectAll: () => void;
+  getUnhandledConsents: () => string[];
+  getRequiredWithoutConsent: () => string[];
+  save: () => ConsentObject;
+};
+
+export const COOKIE_NAME = 'city-of-helsinki-cookie-consents';
+export const COOKIE_EXPIRATION_TIME = 60 * 60 * 24 * 365;
+
+function convertStringArrayToKeyConsentObject(array: string[]): ConsentObject {
+  return array.reduce((current, key) => {
+    // eslint-disable-next-line no-param-reassign
+    current[key] = false;
+    return current;
+  }, {} as ConsentObject);
+}
+
+function mergeConsents(set1: ConsentObject, set2: ConsentObject, set3?: ConsentObject): ConsentObject {
```

Comment:
We might be able to improve the function order in this file. It seems the exported and regular functions are in mixed order currently.